app/src/components/assistant-ui/thread-list.tsx

```
@@ -324,25 +164,29 @@ const HierarchicalThreadList: FC<HierarchicalThreadListProps> = ({
     const updateThreadAction = useAction(api.chat.functions.updateThread);
-
-    // Thread order mutation
     const updateThreadOrderMutation = useMutation(api.chat.functions.updateThreadOrder);
+    const undoDeleteThreadMutation = useMutation(api.chat.functions.undoDeleteThread);
+
+    const handleUndoDeleteThread = useCallback(
+        async (threadId) => {
+            try {
+                await undoDeleteThreadMutation({ threadId });
+                // setCurrentThreadId(threadId); // This line was removed from HierarchicalThreadList props
+            } catch {}
```

ESLint

[error] Empty block statement.

(no-empty)